### PR TITLE
Improve compatibility with google-cloud-pubsub>=2.0.0

### PR DESCRIPTION
**Pull request by mik-laj**
Part of: https://github.com/apache/airflow/issues/12116

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Static checks failing :(

File: airflow/providers/google/cloud/hooks/pubsub.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ class PubSubHook(GoogleBaseHook):
     Hook for accessing Google Pub/Sub.
 
     The Google Cloud project against which actions are applied is determined by
-    the project embedded in the Connection referenced by gcp_conn_id.
+    the project embedded in the Connection referencqed by gcp_conn_id.
```
Comment (TobKed) [inline airflow/providers/google/cloud/hooks/pubsub.py]: typo

**Review by TobKed — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

File: setup.py [hunk 2]
```diff
@@ -266,7 +266,7 @@ def write_version(filename: str = os.path.join(*[my_dir, "airflow", "git_version
     'google-cloud-memcache>=0.2.0',
     'google-cloud-monitoring>=0.34.0,<2.0.0',
     'google-cloud-os-login>=2.0.0,<3.0.0',
-    'google-cloud-pubsub>=1.0.0,<2.0.0',
+    'google-cloud-pubsub>=2.0.0,<3.0.0',
```
Comment (potiuk) [inline setup.py:269]: Should we change those two lines above to `~2.0`  ?

Comment (mik-laj) [reply setup.py:269]: How will it benefit us? In the remaining lines, I have one convention and I think it's worth keeping one style.

**Pull request merged**